src/aws/nodemanager/nodemanager.py

```diff
@@ -71,18 +99,34 @@ def generate_heartbeat(self, notify=True):
 
     def process_heartbeat(self, hb, source) -> Packet:
         hb['source'] = source  # Set the source IP of the heartbeat (i.e., the worker).
-
+        self.notify(hb)  # Forward the heartbeat to the monitor for metrics.
+        if len(self._tasks)>0:  #If there are tasks in the taskpool send a new command to the worker
+            packet=CommandPacket(command="task")
+            current_task=self._tasks.pop(0)
+            packet["task_data"]= current_task.get_task_data()
```

Comment:
For now, this is okay. We need to think about how we divide the work over workers. In this setup you just give work to the first worker giving a heartbeat if there is work to divide. Later we should make sure to give the task to the worker having the least work or at least make sure that not all tasks are simply sent to the same worker.